include/boost/url/detail/remove_dot_segments.hpp

```diff
@@ -0,0 +1,149 @@
+//
+// Copyright (c) 2022 alandefreitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_DETAIL_REMOVE_DOT_SEGMENTS_HPP
+#define BOOST_URL_DETAIL_REMOVE_DOT_SEGMENTS_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/string_view.hpp>
+
+namespace boost {
+namespace urls {
+namespace detail {
+
+/** Return a string with remove_dot_segments applied.
```

Comment:
we don't put full javadocs on non-public interfaces..

Reply:
I thought this was going to be public at first so I wrote the javadocs. So I didn't see the point in removing it when I made it a detail.

Reply:
There are reasons to remove it. First, I don't make the mistake of reviewing it needlessly since it looks public. Second, we don't have to maintain it when it changes. And third, since it is removed it can never become out of sync.